pallets/nfts/src/tests.rs

```diff
@@ -4112,169 +4129,116 @@ fn clear_collection_metadata_works() {
 }
 
 #[test]
-fn collection_item_works() {
+fn mint_requires_deposit_should_works() {
 	new_test_ext().execute_with(|| {
-		let collection_id = 0;
-		let total_items = 10;
-		let user_id = account(1);
-
-		// No collection.
-		assert_eq!(Nfts::collection_items(collection_id), None);
-
+		Balances::make_free_balance_be(&account(1), 100);
+		Balances::make_free_balance_be(&account(2), 100);
 		assert_ok!(Nfts::force_create(
 			RuntimeOrigin::root(),
-			user_id.clone(),
-			default_collection_config()
+			account(1),
+			collection_config_with_all_settings_enabled()
 		));
+		// Minting reserves deposit from the collection owner.
+		assert_ok!(Nfts::mint(RuntimeOrigin::signed(account(1)), 0, 42, account(1), None));
+		assert_eq!(AccountBalance::<Test>::get(0, account(1)), Some((1, (account(1), 1))));
+		assert_eq!(Balances::reserved_balance(&account(1)), 1 + 1); // 1 - item deposit, 1 - balance deposit
+		assert_eq!(Nfts::owner(0, 42).unwrap(), account(1));
 
-		// Mint items and validate the total supply.
-		(0..total_items).into_iter().for_each(|i| {
-			assert_ok!(Nfts::force_mint(
-				RuntimeOrigin::root(),
-				collection_id,
-				i,
-				user_id.clone(),
-				ItemConfig::default()
-			));
-		});
-		assert_eq!(Nfts::collection_items(collection_id), Some(total_items));
+		// Minting for accounts with a non-zero balance requires no additional reserves.
+		assert_ok!(Nfts::mint(RuntimeOrigin::signed(account(1)), 0, 43, account(1), None));
+		assert_eq!(AccountBalance::<Test>::get(0, account(1)), Some((2, (account(1), 1))));
+		assert_eq!(Balances::reserved_balance(&account(1)), 2 + 1); // 2 - item deposit, 2 - balance deposit
+		assert_eq!(Nfts::owner(0, 43).unwrap(), account(1));
+
+		// Increase the reserved balance on minting for a new `AccountBalance` record.
+		assert_ok!(Nfts::mint(RuntimeOrigin::signed(account(1)), 0, 44, account(2), None));
+		assert_eq!(AccountBalance::<Test>::get(0, account(2)), Some((1, (account(1), 1))));
+		assert_eq!(Balances::reserved_balance(&account(1)), 3 + 2); // 3 - item deposit, 2 - balance deposit
+		assert_eq!(Nfts::owner(0, 44).unwrap(), account(2));
+
+		// Minting by an admin.
+		assert_ok!(Nfts::set_team(
+			RuntimeOrigin::signed(account(1)),
+			0,
+			Some(account(2)),
+			Some(account(3)),
+			Some(account(4)),
+		));
+		assert_ok!(Nfts::mint(RuntimeOrigin::signed(account(2)), 0, 45, account(5), None));
+		assert_eq!(AccountBalance::<Test>::get(0, account(5)), Some((1, (account(2), 1))));
+		assert_eq!(Balances::reserved_balance(&account(2)), 1 + 1); // 1 - item deposit, 1 - balance deposit
+		assert_eq!(Nfts::owner(0, 45).unwrap(), account(5));
+
+		assert_eq!(collections(), vec![(account(1), 0)]);
+		assert_eq!(
+			items(),
+			vec![
+				(account(1), 0, 42),
+				(account(1), 0, 43),
+				(account(2), 0, 44),
+				(account(5), 0, 45)
+			]
+		);
 	});
 }
 
 #[test]
-fn clear_collection_approvals_works() {
```

Comment:
Moved to the bottom.

Reply:
Please revert for this PR as it creates noise and it is not important for this PR